tests/test_drift.py

```diff
@@ -52,7 +52,7 @@ def test_device_like_torch_module():
     if not torch.cuda.is_available():
         return
 
-    element = cheetah.Drift(length=torch.tensor(0.2), device="cuda")
+    element = cheetah.Drift(length=torch.tensor([0.2]), device="cuda")
```

Comment:
I understand it's difficult for broadcasting... but it seems really annoying that we have to do `length=torch.tensor([0.1])` for each element.

I would suggest that at least for default single lattice (non-batched) tracking we should allow users to just define a magnet without specifying the shape of the tensor..?  (e.g. provide some broadcasting from [] to [1]...?)

Reply:
Or...maybe we mention it explicitly in the documentation that one can do. This is probably a easier way than putting `torch.tensor([...])` everywhere?
```python
segment = segment.broadcast(incoming_beam_batch_shape)
out_beam = segment.track(incoming_beam)
```

Reply:
I agree. PyTorch actually does similar automatic broadcasting. But I would wait to integrate that until we know the computations and the general batched/vectorised design works.

Reply:
So make sure it works first, then make sure it's nice to use. Right now I'm still chasing down at least one bug in the vectorised code.